### ISSUE TITLE
[WASM] Icon in NavigationViewItem flickers when selection is toggled

### ISSUE THREAD

**Issue by jcantin-nventive** [unoplatform/uno#3520]
## Current behavior

In a NavigationView, when there is a style which modifies the Foreground of a BitMapIcon on selection, there is a flicker of the icon, both for the newly selected item and the old selected item.

## Expected behavior

In a NavigationView, when there is a style which modifies the Foreground of a BitMapIcon on selection, there is no flicker of the icon, it simply changes colours.

## How to reproduce it (as minimally and precisely as possible)

https://github.com/jcantin-nventive/JCantinUnoTestApp

Launch the app and select one item, then the other.

## Environment

Nuget Package: Uno.UI

Package Version(s): 3.0.0-dev.758

Affected platform(s):

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [x] 2017 (version: 15.9.11)
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by MartinZikmund** [comment]
I think I saw similar problem for `TabView` as well

**Comment by MartinZikmund** [comment]
I can confirm this is still happening:

![bug](https://user-images.githubusercontent.com/1075116/123541030-783cdc80-d742-11eb-9c3b-f3791ddc5823.gif)

In my case the code looks as follows:

```
<mux:NavigationViewItem>
    <mux:NavigationViewItem.Icon>
        <BitmapIcon UriSource="ms-appx:///Assets/NavigationViewLogo.png" ShowAsMonochrome="False" />
    </mux:NavigationViewItem.Icon>
</mux:NavigationViewItem>
```

**Comment by Youssef1313** [comment]
Will likely get fixed with https://github.com/unoplatform/uno/pull/6726

**Comment by francoistanguay** [comment]
Is this fixed @Youssef1313  @MartinZikmund  ?

**Comment by Youssef1313** [comment]
@francoistanguay It should be fixed by #6726. @MartinZikmund Can you confirm?

**Comment by MartinZikmund** [comment]
Fixed!
